### PR TITLE
Replace Standard Answer Categories with Assistant assignment

### PR DESCRIPTION
**Pull request by trial-danswer**
## Description
This PR removes the concept of Standard Answer Categories and replaces it with an alternative method of choosing when to trigger Standard Answer responses. 

A Standard Answer can now be assigned to an Assistant upon creation or editing, or configured as a "global" Standard Answer that responders (Slack bot, web queries, chat interaction) can select. 

A new `get_standard_answers_for_personas_or_global(persona_ids)` convenience function is now available in `ee/danswer/db` to get a list of answers that match this criteria.

## Screenshots
image
Screenshot 2024-09-16 at 6 28 17 PM

## How Has This Been Tested?
Locally

## Accepted Risk
TBD

## Related Issue(s)
[If applicable, link to the issue(s) this PR addresses]

## Checklist:
- [ ] All of the automated tests pass
- [ ] All PR comments are addressed and marked resolved
- [ ] If there are migrations, they have been rebased to latest main
- [ ] If there are new dependencies, they are added to the requirements
- [ ] If there are new environment variables, they are added to all of the deployment methods
- [ ] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [ ] Docker images build and basic functionalities work
- [ ] Author has done a final read through of the PR right before merge

### CODE REVIEW

File: web/src/app/ee/admin/standard-answer/lib.ts [hunk 1]
```diff
@@ -5,53 +5,22 @@ export interface StandardAnswerCategoryCreationRequest {
 export interface StandardAnswerCreationRequest {
   keyword: string;
   answer: string;
-  categories: number[];
   matchRegex: boolean;
   matchAnyKeywords: boolean;
+  applyGlobally: boolean;
+  personas: number[]; // ids
```
Comment (pablonyx) [inline web/src/app/ee/admin/standard-answer/lib.ts]: nit: would prefer personaIds, we use personas in the codebase a lot to refer to the full object

File: backend/alembic/versions/729fa47417b3_standard_answer_categories_to_persona_.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+Revises: 5c7fdadae813
+Create Date: 2024-09-16 12:07:08.615388
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+# revision identifiers, used by Alembic.
+revision = "729fa47417b3"
+down_revision = "5c7fdadae813"
```
Comment (pablonyx) [inline backend/alembic/versions/729fa47417b3_standard_answer_categories_to_persona_.py]: Pretty sure there's an alembic head conflict

File: web/src/components/persona/PersonaDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+              key={persona.id}
+              onClick={() => {
+                handleDeselectPersona(persona);
+              }}
+              className={`
+                  flex 
+                  rounded-lg 
+                  px-2 
+                  py-1 
+                  border
```
Comment (pablonyx) [inline web/src/components/persona/PersonaDropdown.tsx:69]: Would double check the formatting here
Screenshot 2024-09-16 at 6 57 09 PM

File: web/src/app/ee/admin/standard-answer/lib.ts (hunk 1 above)
Comment (trial-danswer) [reply web/src/app/ee/admin/standard-answer/lib.ts]: fixed

File: web/src/components/persona/PersonaDropdown.tsx (hunk 3 above)
Comment (trial-danswer) [reply web/src/components/persona/PersonaDropdown.tsx:69]: The styling is from that component, I didn't reach into it at all. Might be a more complex fix to extract out this problematic styling if the other usages aren't experiencing this.

File: backend/alembic/versions/729fa47417b3_standard_answer_categories_to_persona_.py (hunk 2 above)
Comment (trial-danswer) [reply backend/alembic/versions/729fa47417b3_standard_answer_categories_to_persona_.py]: fixed

**Pull request closed without merging**